### PR TITLE
Backend paging

### LINKED ISSUE #548: Backend - unauthenticated filter requests
**Describe the bug**
At the moment of writing, it is possible to call the filter endpoints without a session key. When that happens, the backend will throw an error, but without the correct cause.

**To Reproduce**
1. `curl -X "GET" "http://localhost:4096/api-osoc/user/filter" -i -d '{  }'  -H "Content-Type: application/json"`
2. See error

**Expected behavior**
The backend throws an unauthenticated request error - right now it throws an internal server error

**Screenshots**
n/a

**Branch** development

**Additional context**
The bug is in the parsers - none of the filter parsers does actually check for the session key, but extracts it when constructing the final return object. We should instead opt to use `const authenticated = await parseKeyRequest(req);` in the beginning of the parser. This also makes it easier to construct the final return object (we can use `...authenticated` to extract the session key from the authenticated, parsed request).

### PR DESCRIPTION
**Pull request by jay-tux**
Fixes #548
Backend part of #524 

- [x] Add pagination-related request types
- [x] Update parsers to contain pagination data
- [x] Update tests for parsers
- [x] Add pagination-related response types
- [x] Update routes to use pagination
- [x] Update tests for routes
- [x] Add authentication to filter request parsers

### CODE REVIEW

File: backend/config.json [hunk 1]
```diff
@@ -85,7 +83,9 @@
         ],
         "preferred": "/api-osoc",
         "authScheme": "auth/osoc2",
-        "defaultUserId": 3
+        "defaultUserId": 3,
+        "port": 4096,
+        "pageSize": 3
```
Comment (Mouwrice) [inline backend/config.json:88]: 3 seems rather small but we can keep it at that for easier testing.

**Review by Mouwrice — APPROVED** [review]

File: backend/config.json (hunk 1 above)
Comment (jay-tux) [reply backend/config.json:88]: Right, gotta change!

**Comment by jay-tux** [comment]
> The backend crashes upon running.

@Mouwrice can you perhaps give me some more info about the crash?

**Comment by Mouwrice** [comment]
> > The backend crashes upon running.
> 
> @Mouwrice can you perhaps give me some more info about the crash?

This was fixed by making types more uniform in PR #573 

But there still are some issues as you can see in the tests.

**Review by Mouwrice — APPROVED** [review]

**Comment by jay-tux** [comment]
@Mouwrice the errors should've been fixed. Can you confirm?

**Comment by Mouwrice** [comment]
> @Mouwrice the errors should've been fixed. Can you confirm?

Yessir!

**Review by KoenDesplenter — APPROVED** [review]
Nice work!

**Pull request merged**